File: 2023/java/src/test/java/org/frieder/aoc/day8/TestDay8A.java
package org.frieder.aoc.day8;

import org.frieder.aoc.day8.a.Solution8A;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay8A {
    @Test
    public void testExample() throws IOException {
        assertEquals(2, Solution8A.getResult("../resources/8/test_a.txt"), 0);
    }

    @Test
    public void testResult() throws IOException {
        assertEquals(18157, Solution8A.getResult("../resources/8/input.txt"), 0);
    }
}

File: 2023/php/test/TestDay8A.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day8\a\Solution8A;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay8A extends TestCase
{

    public function testExample(): void
    {
        $this->assertSame(2, Solution8A::getResult("../resources/8/test_a.txt", TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(18157, Solution8A::getResult("../resources/8/input.txt", TestLogger::createTestLogger()));
    }

}